src/core/server/plugins/discovery/plugin_manifest_parser.test.ts

```diff
@@ -247,72 +247,110 @@ test('return error when manifest contains unrecognized properties', async () =>
   });
 });
 
-describe('requiredOpenSearchPlugins', () => {
-  test('return error when plugin `requiredOpenSearchPlugins` is a string and not an array of string', async () => {
+describe('requiredEnginePlugins', () => {
+  test('return error when plugin `requiredEnginePlugins` is a string and not an object', async () => {
     mockReadFilePromise.mockResolvedValue(
       Buffer.from(
         JSON.stringify({
-          id: 'id1',
+          id: 'invalid-manifest-plugin',
           version: '7.0.0',
           server: true,
-          requiredOpenSearchPlugins: 'abc',
+          requiredEnginePlugins: 'abc',
         })
       )
     );
 
     await expect(parseManifest(pluginPath, packageInfo, logger)).rejects.toMatchObject({
-      message: `The "requiredOpenSearchPlugins" in plugin manifest for "id1" should be an array of strings. (invalid-manifest, ${pluginManifestPath})`,
+      message: `The "requiredEnginePlugins" in plugin manifest for "invalid-manifest-plugin" should be a map of plugin name to its compatible version ranges. (invalid-manifest, ${pluginManifestPath})`,
       type: PluginDiscoveryErrorType.InvalidManifest,
       path: pluginManifestPath,
     });
   });
 
-  test('return error when `requiredOpenSearchPlugins` is not a string', async () => {
+  test('return error when plugin `requiredEnginePlugins` is an array and not an object', async () => {
     mockReadFilePromise.mockResolvedValue(
-      Buffer.from(JSON.stringify({ id: 'id2', version: '7.0.0', requiredOpenSearchPlugins: 2 }))
+      Buffer.from(
+        JSON.stringify({
+          id: 'invalid-manifest-plugin',
+          version: '7.0.0',
+          server: true,
+          requiredEnginePlugins: [{ 'plugin-1': '1.0.0.0', 'plugin-2': '^2.0.1' }],
+        })
+      )
     );
 
     await expect(parseManifest(pluginPath, packageInfo, logger)).rejects.toMatchObject({
-      message: `The "requiredOpenSearchPlugins" in plugin manifest for "id2" should be an array of strings. (invalid-manifest, ${pluginManifestPath})`,
+      message: `The "requiredEnginePlugins" in plugin manifest for "invalid-manifest-plugin" should be a map of plugin name to its compatible version ranges. (invalid-manifest, ${pluginManifestPath})`,
       type: PluginDiscoveryErrorType.InvalidManifest,
       path: pluginManifestPath,
     });
   });
 
-  test('return error when plugin requiredOpenSearchPlugins is an array that contains non-string values', async () => {
+  test('return error when plugin requiredEnginePlugins is an object but contains non-string version value', async () => {
     mockReadFilePromise.mockResolvedValue(
       Buffer.from(
-        JSON.stringify({ id: 'id3', version: '7.0.0', requiredOpenSearchPlugins: ['plugin1', 2] })
+        JSON.stringify({
+          id: 'test-invalid-version-type-plugin',
+          version: '7.0.0',
+          requiredEnginePlugins: { 'invalid-plugin-version-range': 2 },
```

Comment:
I thought `semver` was able to convert this to `^2.0.0`; no?

Reply:
I'm doing `semver.coerce` when reading the version from engine, but this is coming from plugin manifest file and we still expect plugin users to write it in actual `semver` format.  Hence there is a validation check on if the defined version is valid `semver` range or not.